### ISSUE TITLE
Multiple warnings when “CLANG_WARN_OBJC_IMPLICIT_RETAIN_SELF” warning is enabled

### ISSUE THREAD

**Issue by Augustyniak** [rs/SDWebImage#2224]
### New Issue Checklist

* [ x] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [x ] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [x ] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | ios 
 Platform Version        | 11.0
 SDWebImage Version      | 4.3.2
 Integration Method      |  cocoapods 
 Xcode Version           | Xcode 9.2
 Repro rate              | e.g. all the time (100%) 
 Repro with our demo prj | It happens with a demo project but only after minimal modifications are made. "CLANG_WARN_OBJC_IMPLICIT_RETAIN_SELF" warning needs to be enabled in `SDWebImage.xcodeproj`.

### Issue Description and Steps

Xcode shows multiple warnings when "CLANG_WARN_OBJC_IMPLICIT_RETAIN_SELF" warning is enabled in projects settings.

<img width="393" alt="screen shot 2018-02-20 at 4 28 43 pm" src="https://user-images.githubusercontent.com/4410346/36463993-28983848-1682-11e8-83c8-5b5766df19d9.png">

**Comment by dreampiggy** [comment]
These block always need retain `self`. Just replace that with `self`. Through seems no nesscessary, I can change this.

**Comment by dreampiggy** [comment]
Fixed by #2225

**Comment by keith** [comment]
Thanks!

**Comment by Augustyniak** [comment]
That was fast. Thanks!
